src/source-af-packet.c

```diff
@@ -782,6 +782,7 @@ int AFPReadFromRing(AFPThreadVars *ptv)
             SCReturnInt(AFP_FAILURE);
         }
         PKT_SET_SRC(p, PKT_SRC_WIRE);
+        p->pkt_mode = PKT_MODE_AUTO;
```

Comment:
On 12-04-16 08:44, Giuseppe Longo wrote:
> I think that this function should be rework as below:
> 
> |if (PacketModeIsIDS(p)) return 0; else if (PacketModeIsIPS(p)) return 1;
> else /* implied PKT_MODE_AUTO */ return stream_inline; |
> 
> So what we do here is to decide if we do IPS or IDS on the packet, but
> if the pkt_mode is 'AUTO' (like in af-packet)
> we check if stream.inline is enabled or not.
> 

What if 'stream.inline' is set to AUTO itself? Would that work?